### ISSUE TITLE
When creating full 4x4 sudokus, sometimes invalid sudokus are created and sometimes an AssertionError is thrown...

### ISSUE THREAD

**Issue by Lemkinator** [sfuhrm/sudoku#8]
## Expected Behavior
```java
Creator.createFull(GameSchemas.SCHEMA_4X4)
```
 should return a valid 4x4 Sudoku.

## Actual Behavior
```java
Creator.createFull(GameSchemas.SCHEMA_4X4)
```
(1) returns an invalid Sudoku:
Example in Kotlin: 
![invalid sudoku](https://user-images.githubusercontent.com/93702086/203062155-dbe43259-5eba-450f-835e-6b143dc6caeb.png)

<br>

```java
Creator.createFull(GameSchemas.SCHEMA_4X4)
```
(2) throws an AssertionError:
Example in Java:
Arguments: -t -e Both  -s S4x4
JVM-Options: -ea
![assertion error](https://user-images.githubusercontent.com/93702086/203062470-4f1f6965-1b49-4d36-b626-478a13dbb886.png)

## Steps to Reproduce the Problem

  1. Enable AssertionErrors in JVM-Options -> -ea
  2. Call Creator.createFull(GameSchemas.SCHEMA_4X4) multiple times.

**Comment by Lemkinator** [comment]
More Background information:
I'm using this Lib in my Android App and the App sometimes crashes because of that AssertionError. When i catch the AssertionError and just retry creating the sudoku everything works fine, except in some rare cases where an invalid sudoku is created (and no AssertionError is thrown)...

Creating 9x9 or larger Sudokus seems to be fine...

**Comment by sfuhrm** [comment]
Thank you very much for the helpful bug report!
I could reproduce and solve the problem with that. Solution validated with unit test inspired by your reproduce instructions.

Background: In case of a dead-end sudoku, the do-loop in Creator.createFull() did not clear riddle again.
Because of that the new initialization failed.

Fix is in release 5.0.1.
